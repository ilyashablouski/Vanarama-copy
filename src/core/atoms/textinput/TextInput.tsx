import cx from 'classnames';
import React from 'react';
import Text from '../text';
import { ITextInputProps } from './interfaces';

const TextInput: React.FC<ITextInputProps> = React.forwardRef<
  HTMLInputElement,
  ITextInputProps
>((props, ref) => {
  const {
    calculated,
    className,
    dataTestId,
    prefix,
    type = 'text',
    width,
    suffix,
    ...rest
  } = props;

  const style = typeof width !== 'undefined' ? { width } : undefined;
  return (
    <div className={cx('textinput regular', className)} style={style}>
      <input
        {...rest}
        className={cx('textinput--native regular', className, {
          '-calculated': calculated,
          '-prefix': prefix,
          '-suffix': suffix,
        })}
        data-testid={dataTestId}
        ref={ref}
        type={type}
        readOnly={calculated}
      />
      {prefix && (
        <Text className="textinput--prefix" size="regular" color="dark">
          {prefix}
        </Text>
      )}
      {suffix && (
        <Text className="textinput--suffix" size="regular" color="dark">
          {suffix}
        </Text>
      )}
    </div>
  );
});

TextInput.displayName = 'ForwardRef(TextInput)';
export default React.memo(TextInput);
