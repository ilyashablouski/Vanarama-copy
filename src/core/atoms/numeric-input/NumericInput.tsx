import React from 'react';
import TextInput from '../textinput';
import { INumericInputProps } from './interfaces';

const NUMBER_REGEX = /[0-9]/;

const NumericInput: React.FC<INumericInputProps> = React.forwardRef<
  HTMLInputElement,
  INumericInputProps
>(({ onKeyPress, ...rest }, ref) => (
  <TextInput
    {...rest}
    onKeyPress={event => {
      if (!NUMBER_REGEX.test(event.key)) {
        event.preventDefault();
      } else {
        onKeyPress?.(event);
      }
    }}
    ref={ref}
  />
));

NumericInput.displayName = 'ForwardRef(NumericInput)';
export default NumericInput;
