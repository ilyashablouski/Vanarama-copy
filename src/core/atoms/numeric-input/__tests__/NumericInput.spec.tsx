import React from 'react';
import { mount } from 'enzyme';
import TextInput from '../../textinput';
import NumericInput from '../NumericInput';

const getComponent = () => {
  return <NumericInput id="accountNumber" />;
};

const buildEventAndSetSpy = (key: any) => {
  const event = { key, preventDefault: () => {} };
  jest.spyOn(event, 'preventDefault');

  return event;
};

const simulateKeyPress = (wrapper: any, event: any) => {
  const input = wrapper
    .find(TextInput)
    .at(0)
    .find('input');

  input.simulate('keypress', event);
};

describe('<NumericInput />', () => {
  let wrapper: any;

  beforeEach(() => {
    wrapper = mount(getComponent());
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('lets you type numbers', () => {
    const event = buildEventAndSetSpy(3);

    simulateKeyPress(wrapper, event);

    expect(event.preventDefault).toHaveBeenCalledTimes(0);
  });

  it('does not let you type non-numbers', () => {
    const event = buildEventAndSetSpy('a');

    simulateKeyPress(wrapper, event);

    expect(event.preventDefault).toHaveBeenCalledTimes(1);
  });
});
