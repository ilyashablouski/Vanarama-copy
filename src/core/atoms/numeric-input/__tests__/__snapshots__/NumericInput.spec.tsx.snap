// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NumericInput /> renders correctly 1`] = `
<div>
  <div
    class="textinput regular"
  >
    <input
      class="textinput--native regular"
      id="accountNumber"
      placeholder="accountNumber"
      type="text"
    />
  </div>
</div>
`;
